x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/detection_rules_client/detection_rules_client.create_custom_rule.test.ts

```diff
@@ -28,14 +29,15 @@ describe('DetectionRulesClient.createCustomRule', () => {
   let detectionRulesClient: IDetectionRulesClient;
 
   const mlAuthz = (buildMlAuthz as jest.Mock)();
+  let actionsClient: jest.Mocked<ActionsClient>;
 
   beforeEach(() => {
     jest.resetAllMocks();
 
     rulesClient = rulesClientMock.create();
     rulesClient.create.mockResolvedValue(getRuleMock(getQueryRuleParams()));
 
-    detectionRulesClient = createDetectionRulesClient(rulesClient, mlAuthz);
+    detectionRulesClient = createDetectionRulesClient(actionsClient, rulesClient, mlAuthz);
```

Comment:
Does it make sense here to add a test that checks if the system actions passed in the `action` array of the `params` gets created successfully? Something like:
```ts
    const params = {...getCreateRulesSchemaMock(), actions: [/* whatever system action */]};

    await detectionRulesClient.createCustomRule({ params });

    expect(rulesClient.create).toHaveBeenCalledWith(
      expect.objectContaining({
        data: expect.objectContaining({
          enabled: true,
          systemActions: /* whatever system action */],
          params: expect.objectContaining({
            description: params.description,
            immutable: false,
          }),
        }),
      })
    );
```
And the same for patch, update, and import.

Reply:
Good call to add more tests it helped find some logic bugs. This should be fixed now.